src/query/service/src/interpreters/interpreter_insert_v2.rs

```diff
@@ -101,6 +117,177 @@ impl InsertInterpreterV2 {
         let cast_needed = select_schema != *output_schema;
         Ok(cast_needed)
     }
+
+    fn apply_stage_options(
+        &self,
+        stage: &mut UserStageInfo,
+        params: &BTreeMap<String, String>,
+    ) -> Result<()> {
+        for (k, v) in params.iter() {
+            match k.as_str() {
```

Comment:
may we can reuse the parser combinator to do it.
which make it easier to maintain (in one place).

search for `tokenize_sql` in code base, you can see several  code snippets like this.


```
        let tokens = tokenize_sql(sql)?;
        let backtrace = Backtrace::new();
        let exprs = some_parser(&tokens[1..tokens.len()], sql_dialect, &backtrace)?;
```